test/bdd/pkg/issuecredential/issuecredential_sdk_steps.go

```diff
@@ -0,0 +1,258 @@
+	s.Step(`^"([^"]*)" sends an offer to the "([^"]*)"$`, a.sendsOffer)
+	s.Step(`^"([^"]*)" accepts an offer and sends a request to the Issuer`, a.acceptOffer)
+	s.Step(`^"([^"]*)" accepts credential with name "([^"]*)"$`, a.acceptCredential)
+	s.Step(`^"([^"]*)" checks that credential is being stored under "([^"]*)" name$`, a.checkCredential)
+}
+
+func (a *SDKSteps) checkCredential(agent, name string) error {
+	return retry(retryAttempts, time.Second, func() error {
+		store, err := verifiableStore.New(a.bddContext.AgentCtx[agent])
+		if err != nil {
+			return err
+		}
+
+		ID, err := store.GetCredentialIDByName(name)
+		if err != nil {
+			return err
+		}
+
+		_, err = store.GetCredential(ID)
+		return err
+	})
+}
+
+func retry(attempts int, sleep time.Duration, fn func() error) error {
+	if err := fn(); err != nil {
+		attempts--
+		if attempts > 0 {
+			time.Sleep(sleep)
+			return retry(attempts, sleep, fn)
+		}
+
+		return err
+	}
+
+	return nil
+}
+
+func (a *SDKSteps) sendsOffer(agent1, agent2 string) error {
+	if err := a.createClient(agent1); err != nil {
+		return err
+	}
+
+	if err := a.createClient(agent2); err != nil {
+		return err
+	}
+
+	conn, err := a.getConnection(agent1, agent2)
+	if err != nil {
+		return err
+	}
+
+	return a.clients[agent1].SendOffer(&protocol.OfferCredential{}, conn.MyDID, conn.TheirDID)
+}
+
+func (a *SDKSteps) sendsProposal(agent1, agent2 string) error {
+	if err := a.createClient(agent1); err != nil {
+		return err
+	}
+
+	if err := a.createClient(agent2); err != nil {
+		return err
+	}
+
+	conn, err := a.getConnection(agent1, agent2)
+	if err != nil {
+		return err
+	}
+
+	return a.clients[agent1].SendProposal(&protocol.ProposeCredential{}, conn.MyDID, conn.TheirDID)
+}
+
+func (a *SDKSteps) sendsRequest(agent1, agent2 string) error {
+	if err := a.createClient(agent1); err != nil {
+		return err
+	}
+
+	if err := a.createClient(agent2); err != nil {
+		return err
+	}
+
+	conn, err := a.getConnection(agent1, agent2)
+	if err != nil {
+		return err
+	}
+
+	return a.clients[agent1].SendRequest(&protocol.RequestCredential{}, conn.MyDID, conn.TheirDID)
+}
+
+func (a *SDKSteps) acceptProposal(agent string) error {
+	PIID, err := a.getAction(agent)
+	if err != nil {
+		return err
+	}
+
+	return a.clients[agent].AcceptProposal(PIID, &protocol.OfferCredential{})
+}
+
+func (a *SDKSteps) acceptRequest(agent string) error {
+	PIID, err := a.getAction(agent)
+	if err != nil {
+		return err
+	}
+
+	return a.clients[agent].AcceptRequest(PIID, &protocol.IssueCredential{
+		CredentialsAttach: []decorator.Attachment{
+			{Data: decorator.AttachmentData{JSON: getVCredential()}},
+		},
+	})
+}
+
+func (a *SDKSteps) getAction(agent string) (string, error) {
+	select {
+	case <-a.actions[agent]:
+	case <-time.After(time.Second):
+		return "", errors.New("timeout")
+	}
+
+	actions, err := a.clients[agent].Actions()
+	if err != nil {
+		return "", err
+	}
+
+	return actions[0].PIID, nil
```

Comment:
we are always getting first element from the action list. How can a client tie up proposal-credential message to an issue-credential message ?

Reply:
Changed! Now I am receiving an event from the channel.